tests/foreman/cli/test_discoveryrule.py

```diff
@@ -318,178 +323,50 @@ def test_positive_delete(self, discoveryrule_factory):
+        """Update discovery rule parameters
 
         :id: 1045e2c4-e1f7-42c9-95f7-488fc79bf70b
 
-        :expectedresults: Rule name is updated
+        :expectedresults: Rule params are updated
 
         :CaseLevel: Component
 
         :CaseImportance: Medium
         """
-        rule = discoveryrule_factory()
-        new_name = gen_string('numeric')
-        DiscoveryRule.update({'id': rule.id, 'name': new_name})
-        rule = Box(DiscoveryRule.info({'id': rule.id}))
-        assert rule.name == new_name
-
-    @pytest.mark.tier2
-    def test_positive_update_org_loc_by_id(self, discoveryrule_factory):
-        """Update org and location of selected discovery rule using org/loc ids
-
-        :id: 26da79aa-30e5-4052-98ae-141de071a68a
-
-        :expectedresults: Rule was updated and with given org & location.
-
-        :BZ: 1377990
-
-        :CaseLevel: Component
-        """
-        new_org = Box(make_org())
-        new_loc = Box(make_location())
-        new_hostgroup = Box(
-            make_hostgroup({'organization-ids': new_org.id, 'location-ids': new_loc.id})
-        )
-        rule = discoveryrule_factory()
-        DiscoveryRule.update(
-            {
-                'id': rule.id,
-                'organization-ids': new_org.id,
-                'location-ids': new_loc.id,
-                'hostgroup-id': new_hostgroup.id,
-            }
-        )
-        rule = Box(DiscoveryRule.info({'id': rule.id}))
-        assert new_org.name in rule.organizations
-        assert new_loc.name in rule.locations
-
-    @pytest.mark.tier3
-    def test_positive_update_org_loc_by_name(self, discoveryrule_factory):
-        """Update org and location of selected discovery rule using org/loc
-        names
-
-        :id: 7a5d61ac-6a2d-48f6-a00d-df437a7dc3c4
-
-        :expectedresults: Rule was updated and with given org & location.
-
-        :BZ: 1377990
-
-        :CaseLevel: Component
-
-        :CaseImportance: Medium
-        """
-        new_org = Box(make_org())
-        new_loc = Box(make_location())
+        rule = discoveryrule_factory(options={'hosts-limit': '5'})
+        new_name = gen_string('alpha')
+        new_query = 'model = KVM'
+        new_hostname = gen_string('alpha')
+        new_limit = '10'
         new_hostgroup = Box(
-            make_hostgroup({'organization-ids': new_org.id, 'location-ids': new_loc.id})
+            target_sat.cli_factory.make_hostgroup({'organization-ids': class_org.id})
         )
-        rule = discoveryrule_factory()
-        DiscoveryRule.update(
+        target_sat.cli.DiscoveryRule.update(
             {
                 'id': rule.id,
-                'organizations': new_org.name,
-                'locations': new_loc.name,
-                'hostgroup-id': new_hostgroup.id,
+                'name': new_name,
+                'search': new_query,
+                'hostgroup': new_hostgroup.name,
+                'hostname': new_hostname,
```

Comment:
Nice, but this test still doesn't cover `test_positive_update_org_loc_by_id` and `test_positive_update_priority`, can we add that too?

Reply:
Priority and org/loc are being tested individually

Reply:
@adarshdubey-star yes, but can't those be merged with this one?

Reply:
I tried to keep it that way.